### PR TITLE
Add possibility to update branch Model Condition

### PR DESCRIPTION
**Pull request by j0nathan33**

### CODE REVIEW

**Comment by cdancy** [comment]
Please rebase off of master to fix conflicts.

File: src/main/java/com/cdancy/bitbucket/rest/domain/branch/Type.java [hunk 1]
```diff
@@ -17,24 +17,41 @@
...
+
+    public abstract TypeId id();
 
+    @Nullable
     public abstract String displayName();
 
     public abstract String prefix();
 
+    @Nullable
+    public abstract Boolean enabled();
```
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/domain/branch/Type.java]: Can this really be Null? If not lets use unboxed boolean here and below in the `create` method.

Comment (j0nathan33) [inline src/main/java/com/cdancy/bitbucket/rest/domain/branch/Type.java]: I check with api and It return some null value.

Comment (j0nathan33) [inline src/main/java/com/cdancy/bitbucket/rest/domain/branch/Type.java]: I check with api and It return some null value.

**Comment by cdancy** [comment]
All tests work as expected. Thanks again @j0nathan33 !

**Pull request merged**